katana/topo.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import os
+
+import numpy as np
+from netCDF4 import Dataset
+
+
+class Topo():
+    """Class for working with and storing topo information
+    """
+
+    def __init__(self, config):
+        """Init the topo class
+
+        Arguments:
+            config {dict} -- dictionary of configuration options
+        """
+
+        self._logger = logging.getLogger(__name__)
+
+        self.config = config
+        self.topo_filename = self.config['topo']['filename']
+        self.zone_letter = self.config['topo']['zone_letter']
+        self.zone_number = self.config['topo']['zone_number']
+
+        # prefix that wind ninja will use in the file naming convention
+        self.windninja_prefix = os.path.splitext(
+            os.path.basename(self.topo_filename))[0]
+
+        self.windnina_filenames = '{}{}'.format(
+            self.windninja_prefix,
+            self.config['topo']['wind_ninja_topo_suffix']
+        )
+
+        # write ascii dem for WindNinja
+        dir_topo = os.path.dirname(self.topo_filename)
+        self.windninja_topo = os.path.join(dir_topo, '{}.asc'.format(
+            self.windnina_filenames
+        ))
+
+        self.windninja_topo_prj = os.path.join(dir_topo, '{}.prj'.format(
+            self.windnina_filenames
+        ))
+
+        # get info about model domain
+        self.get_topo_stats()
+        self.x1 = self.topo_stats['x']
+        self.y1 = self.topo_stats['y']
+
+        # write new files
+        self.netcdf_dem_to_ascii()
+
+        self._logger.debug('Topo initialized')
+
+    def netcdf_dem_to_ascii(self):
+        """
+        Write a geotagged ascii dem for use with WindNinja
+        Writes the ascii and prj files
+
+        Returns:
+            None
+        """
+        # check if files exist first
+        if os.path.exists(self.windninja_topo):
+            self._logger.warning(
+                '{} already exists, not creating again'.format(
+                    self.windninja_topo))
+
+        # if not, make the files
+        else:
+            fillval = -9999
+            # get the prj file name
+            asc_dir = os.path.dirname(self.windninja_topo)
+            bn = os.path.splitext(os.path.basename(self.windninja_topo))[0]
+            fp_prj = os.path.join(asc_dir, bn+'.prj')
+
+            # get the netcdf
+            ds = Dataset(self.topo_filename, 'r')
+            dem = ds.variables['dem'][:]
+            # flip dem since we are now indexing from the bottom left
+            dem = np.flipud(dem)
+
+            # create header for projection
+            if hasattr(ds.variables['dem'], 'grid_mapping'):
+                gridmap = ds.variables['dem'].grid_mapping
+                prj_head = ds.variables[gridmap].spatial_ref
+            else:
+                self._logger.error('No projection info in topo file')
+
+            # close the netcdf
+            ds.close()
```

Comment:
I know science tends to comment a lot of code and I personally think a lot fo the times it is not much different of what the code already tells. An interesting short read is [this SO thread](https://softwareengineering.stackexchange.com/questions/51307/self-documenting-code-vs-commented-code) and the book Clean Code is a classic on this topic (first answer).

Reply:
Makes sense, I've tried to make all the variable names more descriptive to help not only tracking for me but then it's in a way self describing